From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alistair Delva <adelva@google.com>
Date: Fri, 11 Oct 2019 10:24:23 -0400
Subject: ANDROID: Fix x86_64 allmodconfig build

CONFIG_KVM_INTEL depends on asm volatile goto which is about to be
supported in clang, but we haven't landed the upgraded compiler yet.
Suppress use of this kernel feature for now.

In file included from arch/x86/kvm/vmx/vmx.h:11:
arch/x86/kvm/vmx/ops.h:157:2: error: 'asm goto' constructs are not
supported yet
vmx_asm2(vmwrite, "r"(field), "rm"(value), field, value);
^

Change-Id: If21ca65a7bd1e43c5f66a93a6108cf8aa49a9c64
Signed-off-by: Alistair Delva <adelva@google.com>
---
 build.config.allmodconfig | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/build.config.allmodconfig b/build.config.allmodconfig
index deaa8f3d5a69..854a8f8aa3f2 100644
--- a/build.config.allmodconfig
+++ b/build.config.allmodconfig
@@ -2,12 +2,14 @@ DEFCONFIG=allmodconfig
 KCONFIG_ALLCONFIG=${ROOT_DIR}/common/arch/${ARCH%_*}/configs/gki_defconfig
 
 # XFS_FS is currently broken on this branch with clang-9
+# KVM_INTEL is broken on this branch due to lack of asm-goto support in clang
 POST_DEFCONFIG_CMDS="update_config"
 function update_config() {
     ${KERNEL_DIR}/scripts/config --file ${OUT_DIR}/.config \
          -d TEST_KMOD  \
          -d XFS_FS \
-         -d CPU_BIG_ENDIAN
+         -d CPU_BIG_ENDIAN \
+         -d KVM_INTEL
     (cd ${OUT_DIR} && \
      make O=${OUT_DIR} $archsubarch CC=${CC} CROSS_COMPILE=${CROSS_COMPILE} olddefconfig)
 }
